lib/EarlyFunctionAnalysis/AttachDebugInfo.cpp

```diff
@@ -74,78 +78,102 @@ static bool isTrue(const llvm::Value *V) {
   return getLimitedValue(V) != 0;
 }
 
-static void handleFunction(DIBuilder &DIB,
-                           llvm::Function &F,
-                           DISubprogram *TheSubprogram,
-                           efa::ControlFlowGraph &FM,
-                           GeneratedCodeBasicInfo &GCBI) {
-  namespace ranks = revng::ranks;
+class DebugInformationBuilder {
+  DIBuilder &DIB;
```

Comment:
1. Add `private:`
2. Zero/`nullptr` initialize all the pointers, enums, scalars.